src/utils.js

```diff
@@ -209,3 +211,29 @@ export function createModelledFunctionCall(
     }
   };
 }
+
+export function bindingWasMutated(binding: Binding, effects: Effects, F: FunctionValue): boolean {
```

Comment:
The function isn't quite finding out if a binding was mutated, it's supposed to determine whether a ModifiedBinding mutates nonlocal state, so maybe `isMutationOutsideFunction` or `mutatesNonLocalState`

Reply:
Conceptually the original approach was: if the binding was created by an environment created by an optimized function (or any of its nested functions), then don't emit this.
The new approach is: if the binding was created in a parent scope of the optimized function, emit it.

These are equivalent iff it's impossible to have a modified binding that is neither 
1. created by a parent scope of the optimized function 
2. created by an environment nested within an optimized function
Which I'm not convinced is true (although it may be for pure functions). 

Looks like this is a test case that fails:
```javascript
let outermost;
  function foo() {
    let inner1 = 0;
    return [() => inner1, () => inner1++];
  }
  (function() {
    let outer;
    let [getter, incrementer] = foo();
    function toOptimize() {
      let inner2;
      incrementer();
      return getter();
    }
    global.bar = toOptimize;
    global.getter = getter;
    global.__optimize && global.__optimize(toOptimize);
  })();

  global.inspect = function() {
    let x = global.getter();
    let y = global.bar();
    let z = global.getter();
    return x + " " + y + " " + z;
  }
```

Here the environment nesting/chain is:
global => outermost => inner1
                                    => outer => inner2
Optimized Function's environment is `inner2`, but `inner1` isn't a parent of `inner2`, but mutations to its bindings are observable and need to be serialized.